neqo-transport/src/shuffle.rs

```diff
@@ -121,6 +124,19 @@ mod tests {
         assert!(super::find_sni(truncated).is_none());
     }
 
+    #[test]
+    fn find_sni_invalid_sni_length() {
+        // ClientHello with an SNI extension with an invalid length
+        let mut buf = Vec::from(BUF_WITH_SNI);
+        let len = buf.len();
+
+        assert!(buf[len - 23] == 0x00 && buf[len - 22] == 0x0c); // Check Server Name List length
+                                                                 // Set Server Name List length to 0
```

Comment:
Bad formatting?

Reply:
I'm more concerned about the constants you have littered around.  22 and 23 here, 15 and 39 above.  Still, I can't see how that would be easy to fix.